### DISCUSSION TITLE
Detecting "ground"

### DISCUSSION THREAD

**Discussion by ambs** [VelcroPhysics/VelcroPhysics#68] [Q&A] [answered]
Hi
I am still learning to use VelcroPhysics. I managed to have a boolean on a player being grounded or not, activating or deactivating it on collision/separation. While it works, it allows the player to climb walls.
I see there is a contact object, that I can probably use to find out the position of the collision. Is that possible?
Or the way to go is to use a ray-cast?
Thank you

**Comment by Genbox** [comment, marked as answer]
There are several ways of doing this.

Method 1: You can get the normal of the collision between two shapes. You register the OnCollision event on a body, and in the event method you can use the Manifold property to get the normal. If the normal is 1 (or -1) on the Y axis, then the collision is on the ground. See lines 106 and 215 in this example: https://gist.github.com/Genbox/9b3d2232b6bc22b6c8815487d086c769

Method 2: You can register OnCollision like Method 1 and check if the body in the collision is the ground body. Example here: https://gist.github.com/Genbox/9737d1155efcdd00c9858cf7a492323e

**Reply by ambs** [reply]
I just found (curiously, 10 minutes ago) an example of using a sensor on Box2D - https://www.iforce2d.net/b2dtut/jumpability
Will test all the alternatives and choose one. Thank you for your suggestions!!
